internal-services/catalog/iib-add-fbc-fragment-to-index-image-task.yaml

```diff
@@ -83,49 +83,75 @@ spec:
       script: |
         #!/usr/bin/env bash
         #
+        # shellcheck disable=SC2207,SC2128,SC2178 # requires refactoring to fix these errors
+        # shellcheck disable=SC2005 # false positive as it thinks tekton parameters might be commands
         isfbcOptIn() {
           TMPFILE=$(mktemp)
           PYXIS_URL="https://pyxis.engineering.redhat.com/v1"
 
-          IFS="/" read REGISTRY REPO IMAGE <<< "${1}"
-          IFS=":" read IMAGE TAG <<< "${IMAGE}"
+          IFS="/" read -r REGISTRY REPO IMAGE <<< "${1}"
+          IFS=":" read -r IMAGE TAG <<< "${IMAGE}"
 
           FETCH_URL="${PYXIS_URL}/repositories/registry/${REGISTRY}/repository/${REPO}/${IMAGE}/tag/${TAG}"
 
           # strips the last "/tag" in case $TAG is not set
           [ -z "${TAG}" ] && FETCH_URL=${FETCH_URL%/tag*}
 
-          curl --negotiate -u: "${FETCH_URL}" -o $TMPFILE
+          curl --negotiate -u: "${FETCH_URL}" -o "${TMPFILE}"
           # prints "false" in case .fbc_opt_in entry is missing
-          jq -e -r '.fbc_opt_in //false' $TMPFILE && rm -f $TMPFILE
+          jq -e -r '.fbc_opt_in //false' "${TMPFILE}" && rm -f "${TMPFILE}"
+        }
+
+        # checks if there was any previous build for the same snapshot,
+        # although it should only consider resume it if is not a stage
+        # build as stage builds have no reason for being resumed.
```

Comment:
Why can't stage builds be resumed? I'm curious.

Reply:
Basically because they execute a single step, which is the IIB build. They are not published or signed, although after the conversation in the biggest thread, it might be the case that the staged build  timed out/failed but IIB is still running. The user might want to know what happened, even though it won't do anything else after finish building.

Reply:
the code was changed to allow "resuming" staged indexes as well. PTAL.

Reply:
ok, thanks